docs/trace/building-your-own-exporter.md

```diff
@@ -0,0 +1,51 @@
+# Building your own Exporter
+
+* Exporters should inherit from `ActivityExporter` and implement `ExportAsync`
+  and `ShutdownAsync` methods.
+* Depending on user's choice and load on the application `ExportAsync` may get
+  called concurrently with zero or more activities.
+* Exporters should expect to receive only sampled-in ended activities.
+* Exporters must not throw.
+* Exporters should not modify activities they receive (the same activity may be
+  exported again by different exporter).
+
+```csharp
+class MyExporter : ActivityExporter
+{
+    public override Task<ExportResult> ExportAsync(
+        IEnumerable<Activity> batch, CancellationToken cancellationToken)
+    {
+        foreach (var activity in batch)
+        {
+            Console.WriteLine(
+                $"[{activity.StartTimeUtc:o}] " +
+                $"{activity.DisplayName} " +
+                $"{activity.Context.TraceId.ToHexString()} " +
+                $"{activity.Context.SpanId.ToHexString()}"
+            );
+        }
+
+        return Task.FromResult(ExportResult.Success);
+    }
+
+    public override Task ShutdownAsync(CancellationToken cancellationToken)
+    {
+        return Task.CompletedTask;
+    }
+
+    protected override void Dispose(bool disposing)
+    {
+        // flush the data and clean up the resource
+    }
+}
+```
+
+* Users may configure the exporter similarly to other exporters.
+* You should also provide additional methods to simplify configuration
```

Comment:
Make it clear what exactly we expect from the extension method for the `MyExporter?

Linking to the Extension method used by Zipkin. I prefer this.
Or pasting the content here.

public static TracerProviderBuilder UseZipkinExporter(this TracerProviderBuilder builder, Action<ZipkinExporterOptions> configure = null, Action<ActivityProcessorPipelineBuilder> processorConfigure = null)

Reply:
I agree with you. This section is copied from the current document so I think we want to address it in another PR? (as this PR is focusing on the skeleton instead of actual meat)

Reply:
And this entire doc should be re-written as it doesn't look like something people can easily follow.

Reply:
Agree. I totally forgot that this was just to get the structure ready!